include/deal.II/grid/grid_tools.h

```diff
@@ -2142,6 +2162,453 @@ namespace GridTools
             }
         }
   }
+
+
+
+  namespace internal
+  {
+    namespace ProjectToObject
+    {
+      struct CrossDerivative
```

Comment:
Can you document what this class does?

Reply:
I have added documentation to this and the other internal functions.